src/App.jsx

```diff
@@ -18,6 +19,7 @@ function App() {
           <Route index path="/" element={<Home />} />
           <Route path="/add" element={<Add />} />
           <Route path="/user" element={<User />} />
+          <Route path="/user/create" element={<UserSignUp />} />
```

Comment:
Can you create a Create.jsx under 'routers' folder and return <Create/> for this route?
In the Create component, get userData(using useCurrentUserData under utils folder) and if the userData is undefined, return <UserSignUp/>, else navigate to '/user'. We need to do this because what if a user is already logged in and enters /user/create URL?
Also, it would be great to create a SignUp folder under pages folder and move the UserSignUp under it.